bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/handler/MeteoAlerteHandler.java

```diff
@@ -0,0 +1,220 @@
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        } catch (MalformedURLException e) {
+            logger.warn("Malformed URL in Météo Alerte request : {}", queryUrl);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        }
+    }
+
+    /**
+     * Update the channel from the last Meteo Alerte data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannels(ApiResponse apiResponse) {
+        Arrays.stream(apiResponse.getRecords()).findFirst().ifPresent(record -> {
+            record.getFields().ifPresent(fields -> {
+                updateAlertString(WIND, fields.getVent());
+                updateAlertString(RAIN, fields.getPluieInondation());
+                updateAlertString(STORM, fields.getOrage());
+                updateAlertString(FLOOD, fields.getInondation());
+                updateAlertString(SNOW, fields.getNeige());
+                updateAlertString(HEAT, fields.getCanicule());
+                updateAlertString(FREEZE, fields.getGrandFroid());
+                updateAlertString(AVALANCHE, fields.getAvalanches());
+
+                fields.getDateInsert().ifPresent(date -> updateDate(OBSERVATION_TIME, date));
+                updateState(COMMENT, new StringType(fields.getVigilanceComment()));
+                updateIcon(WIND, fields.getVent());
+                updateIcon(RAIN, fields.getPluieInondation());
+                updateIcon(STORM, fields.getOrage());
+                updateIcon(FLOOD, fields.getInondation());
+                updateIcon(SNOW, fields.getNeige());
+                updateIcon(HEAT, fields.getCanicule());
+                updateIcon(FREEZE, fields.getGrandFroid());
+                updateIcon(AVALANCHE, fields.getAvalanches());
+            });
+        });
+    }
+
+    public void updateIcon(String channelId, String value) {
+        String iconChannelId = channelId + "-icon";
+        if (isLinked(iconChannelId)) {
+            String pictoName = channelId + (!value.isEmpty() ? "_" + value.toLowerCase() : "");
+            byte[] image = getImage("picto" + File.separator + pictoName + ".gif");
+            if (image != null) {
+                RawType picto = new RawType(image, "image/gif");
+                updateState(iconChannelId, picto);
+            }
+        }
+    }
+
+    private byte @Nullable [] getImage(String iconPath) {
+        byte[] data = null;
+        URL url = FrameworkUtil.getBundle(getClass()).getResource(iconPath);
+        logger.trace("Path to icon image resource is: {}", url);
+        if (url != null) {
+            try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+                InputStream is = url.openStream();
+                BufferedImage image = ImageIO.read(is);
+                ImageIO.write(image, "gif", out);
```

Comment:
Since your images are already `gif`s, a simple byte copy should suffice. I can't think of another reason that you would need to parse the image and then write it out again unless there is some weird image metadata you are trying to accomplish with this.

Reply:
Well, I copied the way it was done in the weathercompany binding.